storage/kubernetes/client.go

```diff
@@ -83,14 +84,24 @@ func offlineTokenName(userID string, connID string, h func() hash.Hash) string {
 	return strings.TrimRight(encoding.EncodeToString(hash.Sum(nil)), "=")
 }
 
+const kubeResourceMaxLen = 63
+
+var kubeResourceNameRegex = regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`)
+
 func (cli *client) urlForWithParams(
 	apiVersion, namespace, resource, name string, params url.Values,
-) string {
+) (string, error) {
 	basePath := "apis/"
 	if apiVersion == "v1" {
 		basePath = "api/"
 	}
 
+	if name != "" && (len(name) > kubeResourceMaxLen || !kubeResourceNameRegex.MatchString(name)) {
+		return "", fmt.Errorf(
```

Comment:
This would result in unsanitized user input being logged. (See GitHub code scanning results below)

Also, just an idea: `s/kubernetes resource name is not valid/invalid kubernetes resource name`